### ISSUE TITLE
Shouldn't `scaleDateTime()` work by default with Instant?

### ISSUE THREAD

**Issue by AndreiKingsley** [JetBrains/lets-plot-kotlin#174]
We can use `Instant` as the data type, but we need to add `scaleDatetime()` to format the axis correctly. Why don't we do this automatically when we have Instant as a data type? Including in the case of `yMin`, `middle` etc. It's also a bit strange that the public api lacks something like `scaleColorDatetime`, but we can implement it with
```
Scale(Aes.COLOR, otherOptions = Options(
        mapOf(
            Option.Scale.DATE_TIME to true
        ))
)
```

**Comment by IKupriyanov-HORIS** [comment]
Hi!
Sorry, the issue description is kinda unclear. May I clarify it? Are you talking about the plot legend that uses numbers as labels instead of formatted date/time?

Like in this example:
```kotlin
%use lets-plot
import java.time.Instant.parse

val instants = listOf(
    "2021-01-01T00:00:00Z",
    "2021-02-23T00:00:00Z",
    "2021-04-03T00:00:00Z",
    "2021-05-04T00:00:00Z",
    "2021-10-05T00:00:00Z"
).map(::parse)

val daysData = mapOf<String, Any>(
    "days" to instants,
    "val" to (1..instants.size)
)

val p = letsPlot(daysData) + geomBar() { x = "days"; color = "days" }
p
```

Actual result:
![image](https://user-images.githubusercontent.com/14200189/232910921-a0d3785a-9524-4d03-b574-df29fe4c0497.png)

And you expected this?
![image](https://user-images.githubusercontent.com/14200189/232910982-8dd972c0-94f7-4a16-a30b-75563d43833d.png)

Am I correct?

**Comment by AndreiKingsley** [comment]
Yes!

**Comment by AndreiKingsley** [comment]
Also i want axis formatting when i use, for example, error bar:
`val instants = listOf(
    "2021-01-01T00:00:00Z",
    "2021-01-01T00:04:00Z",
    "2021-01-01T00:00:00Z",
    "2021-01-01T00:00:00Z",
    "2021-01-01T00:00:00Z",
).map(::parse)
val instants2 = listOf(
    "2021-01-01T00:05:00Z",
    "2021-01-01T00:05:00Z",
    "2021-01-01T00:05:00Z",
    "2021-01-01T00:05:00Z",
    "2021-01-01T00:05:00Z",
).map(::parse)
val daysData = mapOf<String, Any>(
    "t1" to instants,
    "t2" to instants2,
    "val" to (1..instants.size)
)

val p = letsPlot(daysData) + geomErrorBar(stat = Stat.identity) {
    x = "val";
    ymin = "t1"
    ymax = "t2"
}
p`
gives
<img width="597" alt="image" src="https://user-images.githubusercontent.com/49092338/232916658-2d4f0782-2a2d-4bfa-b30e-dde0035909ca.png">
but I expect y date-time formatting

**Comment by alshan** [comment]
Fixed in [LPK v4.4.0](https://github.com/JetBrains/lets-plot-kotlin/blob/master/CHANGELOG.md#440---2023-05-10)
